frontend/src/components/RoomList/NewRoomDialog.tsx

```diff
@@ -0,0 +1,186 @@
+import Typography from "@material-ui/core/Typography";
+import { createStyles, makeStyles, Theme, withStyles, WithStyles } from "@material-ui/core/styles";
+import CloseIcon from "@material-ui/icons/Close";
+import { Alert } from "@material-ui/lab";
+import React, { useContext, useState } from "react";
+
+import { Context } from "../../context/appContext";
+import { apiProvider } from "../../services/api";
+
+const styles = (theme: Theme) =>
+    createStyles({
+        root: {
+            margin: 0,
+            padding: theme.spacing(2)
+        },
+
+        closeButton: {
+            position: "absolute",
+            right: theme.spacing(1),
+            top: theme.spacing(1),
+            color: theme.palette.grey[500]
+        }
+    });
+
+const useStyles = makeStyles({
+    input: {
+        width: "90%",
+        fontSize: 15,
+        marginTop: 15
+    },
+
+    label: {
+        fontSize: 15,
+        transform: "translate(10px, 13.5px) scale(1)"
+    },
+
+    addButton: {
+        backgroundColor: "#2a9df4",
+        color: "white",
+        "&:hover, &:focus": {
+            backgroundColor: "#187bcd"
+        }
+    },
+
+    error: {
+        width: "85%",
+        marginLeft: "5%",
+        marginTop: 10
+    }
+});
+
+export interface DialogTitleProps extends WithStyles<typeof styles> {
+    id: string;
+    children: React.ReactNode;
+    onClose: () => void;
+}
+
+const DialogTitle = withStyles(styles)((props: DialogTitleProps) => {
+    const { children, classes, onClose, ...other } = props;
+    return (
+        <MuiDialogTitle disableTypography className={classes.root} {...other}>
+            <Typography variant="h6">{children}</Typography>
+            {onClose ? (
+                <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
+                    <CloseIcon />
+                </IconButton>
+            ) : null}
+        </MuiDialogTitle>
+    );
+});
+
+const DialogContent = withStyles((theme: Theme) => ({
+    root: {
+        padding: theme.spacing(2),
+        paddingLeft: "10%"
+    }
+}))(MuiDialogContent);
+
+const DialogActions = withStyles((theme: Theme) => ({
+    root: {
+        margin: 0,
+        padding: theme.spacing(1)
+    }
+}))(MuiDialogActions);
+
+interface NewRoomDialogProps {
+    open: boolean;
+    setOpen: (state: boolean) => void;
+}
+
+export function NewRoomDialog({ open, setOpen }: NewRoomDialogProps) {
+    const [
+        {
+            sessionStorageContext: { username, email },
+            socket
+        }
+    ] = useContext(Context);
+    const [roomName, setRoomName] = useState("");
+    const [submit, setSubmit] = useState(false);
+    const [apiErrorMessage, setApiErrorMessage] = useState("");
+    const classes = useStyles();
+
+    const setInitialValue = () => {
+        setOpen(false);
+        setRoomName("");
+        setSubmit(false);
+        setApiErrorMessage("");
+    };
+    const onSubmit = () => {
```

Comment:
Well you can clearly see my mental health already went bananas so here we go...

assyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyync